src/execution_plan/execution_plan_build/execution_plan_awareness.c

```
@@ -82,25 +57,23 @@ void ExecutionPlanAwareness_PropagateAwareness
 	while(parent != NULL && parent->plan == op->plan) {
 		// break if no new aliases were added to the awareness table
 		bool short_circuit = true;
 
-		for(int i = 0; i < n; i++) {
-			const char *alias = keys[i];
+		size_t i = 0;
+		char **alias = NULL;
+		while(hashmap_iter(op->awareness, &i, (void **)&alias)) {
 			short_circuit &=
-				HashTableAdd(parent->awareness, (void*)alias, NULL) == DICT_ERR;
+				hashmap_set(parent->awareness, (void*)alias) != NULL;
```

Cppcheck (2.10-2)

[error] Memory leak

(memleak)